### PR TITLE
doc(pallet-fungible): document public api

### PR DESCRIPTION
**Pull request by PraetorP**

### CODE REVIEW

File: pallets/fungible/src/lib.rs [hunk 1]
```diff
@@ -57,13 +119,13 @@ pub mod pallet {
 	pub enum Error<T> {
 		/// Not Fungible item data used to mint in Fungible collection.
 		NotFungibleDataUsedToMintFungibleCollectionToken,
-		/// Not default id passed as TokenId argument
+		/// Not default id passed as TokenId argument.
```
Comment (Maksandre) [inline pallets/fungible/src/lib.rs:122]: What is default id, 0?

File: pallets/fungible/src/lib.rs [hunk 2]
```diff
@@ -103,15 +168,19 @@ pub mod pallet {
 		QueryKind = ValueQuery,
 	>;
 }
-
+/// Handler for fungible assets.
 pub struct FungibleHandle<T: Config>(pallet_common::CollectionHandle<T>);
 impl<T: Config> FungibleHandle<T> {
+	/// Casts [pallet_common::CollectionHandle] into [FungibleHandle].
```
Comment (Maksandre) [inline pallets/fungible/src/lib.rs:174]: What purpose of this and the next one casts?

File: pallets/fungible/src/lib.rs [hunk 3]
```diff
@@ -103,15 +168,19 @@ pub mod pallet {
...
 	pub fn cast(inner: pallet_common::CollectionHandle<T>) -> Self {
 		Self(inner)
 	}
+
+	/// Casts [FungibleHandle] into [pallet_common::CollectionHandle].
 	pub fn into_inner(self) -> pallet_common::CollectionHandle<T> {
 		self.0
 	}
+	/// Returns a mutable reference to the internal [pallet_common::CollectionHandle].
```
Comment (Maksandre) [inline pallets/fungible/src/lib.rs:183]: Can you reveal why we need it?

File: pallets/fungible/src/lib.rs (hunk 1 above)
Comment (PraetorP) [reply pallets/fungible/src/lib.rs:122]: TokenId is a Newtype(but with pub field) for u32 , so yes, it's 0.

File: pallets/fungible/src/lib.rs (hunk 2 above)
Comment (PraetorP) [reply pallets/fungible/src/lib.rs:174]: FungibleHandle is a newtype over CollectionHandle. This is its public API, according by the signature this method should be called new() IMO

File: pallets/fungible/src/lib.rs (hunk 1 above)
Comment (Fahrrader) [reply pallets/fungible/src/lib.rs:122]: Could you add it in?

Comment (PraetorP) [reply pallets/fungible/src/lib.rs:122]: > Could you add it in the doc?
Done.

File: pallets/fungible/src/lib.rs [hunk 4]
```diff
@@ -103,15 +169,19 @@ pub mod pallet {
...
-
+/// Handler for fungible assets.
 pub struct FungibleHandle<T: Config>(pallet_common::CollectionHandle<T>);
 impl<T: Config> FungibleHandle<T> {
+	/// Casts [pallet_common::CollectionHandle] into [FungibleHandle].
 	pub fn cast(inner: pallet_common::CollectionHandle<T>) -> Self {
 		Self(inner)
 	}
+
+	/// Casts [FungibleHandle] into [pallet_common::CollectionHandle].
```
Comment (CertainLach) [inline pallets/fungible/src/lib.rs:180]: rustdoc принимает ссылки без ``, но вообще они нужны в таком контексте (Мы же как код хотим содержимое отображать), и в rust std их тоже ставят: https://doc.rust-lang.org/src/std/lib.rs.html#6
```
[`FungibleHandle`]
```

Comment (PraetorP) [reply pallets/fungible/src/lib.rs:180]: Как я понял , то для второго тоже лучше так сделать  ?!
```
[`pallet_common::CollectionHandle`]
```

File: pallets/fungible/src/lib.rs [hunk 5]
```diff
@@ -441,6 +523,7 @@ impl<T: Config> Pallet<T> {
 		Ok(allowance)
 	}
 
+	/// Transfers of tokens that `from` gave to `spender` to manage, to `to` ID.
```
Comment (Fahrrader) [inline pallets/fungible/src/lib.rs:526]: This might seem a bit convoluted to people, could you rephrase it like "Transfers parts of a token as a non-owner account, with approval of the owner of the token (see {link to set_allowance()}.\n\n# Arguments\n\n- `collection`: ...". The same for `burn_from`. You can borrow the particular descriptions of the arguments of these two from somebody else who's done the token pallets.

Comment (Fahrrader) [reply pallets/fungible/src/lib.rs:526]: Or rather, use the refungible description, it's pretty good without the need to describe the arguments.

File: pallets/fungible/src/lib.rs [hunk 6]
```diff
@@ -523,7 +537,12 @@ impl<T: Config> Pallet<T> {
 		Ok(allowance)
 	}
 
-	/// Transfers of tokens that `from` gave to `spender` to manage, to `to` ID.
+	/// Transfer FT tokens from one account to another.
```
Comment (Fahrrader) [inline pallets/fungible/src/lib.rs:540]: FT already means "Fungible tokens" :) Just refer to them as "fungible tokens"

File: pallets/fungible/src/lib.rs [hunk 7]
```diff
@@ -459,7 +470,9 @@ impl<T: Config> Pallet<T> {
 		));
 	}
 
-	/// Sets the amount of owner tokens that the spender can manage.
+	/// Set allowance for the spender to `transfer` or `burn` owner's tokens.
+	///
+	/// - `amount`: Amount of tokens the spender is allowed to `transfer` or `burn`.
```
Comment (Fahrrader) [inline pallets/fungible/src/lib.rs:475]: Would be good to add argument descriptions for the other fields here, too, for consistency - also, `spender`, for example, is not that obvious.

File: pallets/fungible/src/lib.rs [hunk 8]
```diff
@@ -523,7 +537,12 @@ impl<T: Config> Pallet<T> {
 		Ok(allowance)
 	}
 
-	/// Transfers of tokens that `from` gave to `spender` to manage, to `to` ID.
+	/// Transfer FT tokens from one account to another.
+	/// Same as the [`transfer`] but spender doesn't needs to be an owner of the token pieces.
+	/// The owner should set allowance for the spender to transfer pieces.
+	///	See [`set_allowance`][`Pallet::set_allowance`] for more details.
+	///  
+	///	[`transfer`]: struct.Pallet.html#method.transfer
```
Comment (Fahrrader) [inline pallets/fungible/src/lib.rs:545]: You probably wanted this construct: [`transfer`](Pallet::transfer)

**Review by Fahrrader — APPROVED** [review]

**Review by CertainLach — APPROVED** [review]

**Review by Maksandre — APPROVED** [review]
